src/cmd/sign.js

```diff
@@ -125,6 +156,170 @@ export default function sign(
+    } catch (e) {
+      throw new WebExtError(
+        `Unable to parse ${updateManifestFileName} file located at ` +
+        `${manifestData.applications.gecko.update_url}`);
+    }
+  });
+}
+
+/**
+ * Validates all data needed to generate a new updatemanifest
+ * if some data is missing/corrupt, the manifest can't be generated
+ */
+function prevalidateUpdateManifestParams(manifestData, updateLink) {
+  // ..first validate if the manifest has the right properties
+  // manifestData.applications.gecko.updateUrl
+  // note: `var == null` is equal to
+  // `(variable === undefined || variable === null)`
+  // see: https://stackoverflow.com/questions/2647867
+  if (manifestData == null ||
+      manifestData.applications == null ||
+      manifestData.applications.gecko == null ||
+      manifestData.applications.gecko.update_url == null) {
+    throw new WebExtError(
+      'update-link was passed, but the manifest.json ' +
+      'is missing the .applications.gecko.update_url property. ' +
+      'therefore, no updateManifest could be generated');
+  } else {
+    // check if the update_url property contains a usable URL
+    let parsed = url.parse(manifestData.applications.gecko.update_url);
+    if (parsed == null || parsed.pathname == null) {
+      // the most elaborate flow workaround
+      let applications =
+        manifestData.applications == null ?
+        {gecko: manifestData.applications.gecko} : manifestData.applications;
+      let gecko =
+        applications.gecko == null ?
+        {gecko: applications.gecko} : manifestData.applications.gecko;
+      let update_url =
+        gecko.update_url == null ?
+        gecko.update_url : gecko.update_url; // flow please
+      throw new WebExtError(
+        'Was unable to parse manifest.applications.gecko.update_url ' +
+        'please check this property in your manifest: ' +
+        `${update_url}`);
+    }
+  }
+
+  // manifestData.version
+  if (manifestData.name == null) {
+    throw new WebExtError(
+      'update-link was passed but the manifest.json ' +
+      'is missing the .name property. ' +
+      'therefore, no updateManifest could be generated');
+  }
+
+  // ..lastly check if the passed updateLink parameter is correct
+  if (updateLink.indexOf('{xpiFileName}') === -1) {
+    throw new WebExtError(
+      'Unable to parse --update-link url, please use {xpiFileName} ' +
+      'as a substitute for the XPI file');
+  }
+
+  return true;
+}
+
+/**
+ * Generates a new updateManifest.json file
+ * It will add an entry to a previously fetched updateManifest.json file
+ * for each generated XPI file.
+ * The result will be stored in the artifacts directory
+ */
+// jslint incorrectly reports that this return statement should be a yield.
+// eslint-disable-next-line
+async function createNewUpdateManifest(
+  id: ?string, XPIPath: string, artifactsDir,
+  manifestData: Object, updateLink: string, oldUpdateManifestData: Object
```

Comment:
All arguments should be keyword args to make them more readable and easier to change later on. This is a simple change using ES6 destructors (just add braces):
````javascript
async function createNewUpdateManifest({
  id: ?string, XPIPath: string, artifactsDir, ...
}) {
````